core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1046,6 +1046,10 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
     */
   protected def _cloneTypeImpl: Bundle = {
     assert(Builder.allowReflectiveAutoCloneType, "reflective autoclonetype is disallowed, this should only happen in testing")
+    Builder.deprecated(
+      "The runtime reflection inference for cloneType (_cloneTypeImpl) is deprecated as of Chisel 3.5:  " +
+      "Use the Compiler Plugin to infer cloneType"
+    )
```

Comment:
I thought that sort of deprecation pattern was discouraged -- shouldnt we only be deprecating on major releases, then removing in next major release?

It seems like in 3.5 we are perhaps deprecating running without plugin. then in 3.6 we could remove. I don't think we can just suddenly deprecate something that major on 3.4.4 since users might go from 3.4 to 3.5 directly

Reply:
In dev today (see notes https://github.com/chipsalliance/chisel3/discussions/2055) we discussed a change to deprecation policy that deprecations on minor versions are now the standard with removal in the subsequent major version. Our reasoning is that this enables better development practice, cleaner code, and more agility.

A basic problem we've had for a long time is that we often deprecate and never remove, and sometimes even deprecate things without replacement. The new development flow is that, if an API is to be changed, it is deleted on master and backported to he latest major version. This deletion on master will keep the main codebase cleaner, while also requiring the developer to update all tests *thus ensuring we know what the code migration path looks like and that it is appropriate to deprecate the API*.

The backport will then fail (because it won't be binary compatible), and the developer is then required to deprecate the API on the backport. This has a massive benefit that the internal hacks that are often necessary to maintain some deprecated APIs will only have to be made on the backport branch and will never exist on master.

Finally, it's true that people may sometimes skip minor versions. It will become part of our release notes and announcement notes _which minor version users need to bump to before bumping to the major release_. This requires a bit of additional discipline but I believe the benefits to be worth it.